tests/utils/test_session_token.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime, timedelta
+
+import jwt
+import pytest
+
+from spylib.utils.session_token import (
+    InvalidIssuerError,
+    MismatchedHostError,
+    SessionToken,
+    TokenAuthenticationError,
+)
+
+api_key = "API_KEY"
+api_secret = "API_SECRET"
+now = datetime.now()
+
+
+def get_token():
+    """
+    This function returns the same data over and over again. We don't use a variable
+    as the fixtures will share the memory and overwrite the value, even with proper scopes.
+    """
+    return {
+        "iss": "https://test.myshopify.com/admin",
+        "dest": "https://test.myshopify.com",
+        "aud": api_key,
+        "sub": 1,
+        "exp": (now + timedelta(0, 60)).timestamp(),
+        "nbf": (now - timedelta(0, 60)).timestamp(),
+        "iat": now.timestamp(),
+        "jti": 123,
+        "sid": "abc123",
+    }
+
+
+@pytest.fixture(scope='function', autouse=False)
+def token():
+    return get_token()
+
+
+def generate_auth_header(token):
+    return f'Bearer {jwt.encode(token, api_secret, algorithm="HS256")}'
+
+
+@pytest.mark.asyncio
+async def test_session_token(token):
+    valid_auth_header = generate_auth_header(token)
+    session_token = SessionToken.decode_token_from_header(valid_auth_header, api_key, api_secret)
+
+    assert session_token.iss == token['iss']
+    assert session_token.dest == token['dest']
+    assert session_token.aud == token['aud']
+    assert session_token.sub == token['sub']
+    assert session_token.exp == token['exp']
+    assert session_token.nbf == token['nbf']
+    assert session_token.iat == token['iat']
+    assert session_token.jti == token['jti']
+    assert session_token.sid == token['sid']
```

Comment:
pytest has a lovely dictionary comparison and error reporting so I'd use that instead:
```suggestion
    assert session_token.dict() == token
```